src/RequireCallStack.hs

```diff
@@ -48,9 +52,43 @@ import RequireCallStack.Internal
 -- @since 0.1.0.0
 type RequireCallStack = (HasCallStack, RequireCallStackImpl)
 
--- | Raise an 'ErrorCall' and incur a 'RequireCallStack' constraint while
--- you do so. This
+-- | If you're running into this class, then you need to add
+-- 'RequireCallStack' to your function's signature, or discharge the
+-- constraint using 'provideCallStack'.
+--
+-- @since 0.1.0.0
+type RequireCallStackImpl = ?provideCallStack :: CallStack
+
+-- | The constructor for this type is intentionally not exported
+data CallStack = CallStack
+
+-- | Raise an 'Control.Exception.ErrorCall' and incur a 'RequireCallStack'
+-- constraint while you do so. This
 --
 -- @since 0.1.0.0
 errorRequireCallStack :: RequireCallStack => String -> x
 errorRequireCallStack = error
+
+instance TypeError ('Text "Add RequireCallStack to your function context or use provideCallStack") => IP "provideCallStack" CallStack
```

Comment:
Could you help to understand this line and how it connects to the fact that the error (`Add RequireCallStack to ...`) will be report if missing `RequireCallStack` constraint ?

Reply:
I'm not sure I understand the question, but this error message will be reported whenever the user has a missing `RequireCallStack` constraint.  This code will remind them to add it or to use `provideCallStack` to eliminate the constraint if they don't want to keep propagating the callstack.